sql/src/test/java/io/crate/integrationtests/InformationSchemaTest.java

```diff
@@ -382,6 +384,24 @@ public void testSelectAnalyzersFromRoutines() throws Exception {
     }
 
     @Test
+    public void testSelectFunctionsFromRoutines() throws Exception {
+        execute("create function substract_test(long, long, long) " +
+            "returns long language JAVASCRIPT " +
+            "as 'function substract_test(a, b, c) { return a - b - c; }'");
+        waitForFunctionCreatedOnAll("substract_test", ImmutableList.of(DataTypes.LONG, DataTypes.LONG, DataTypes.LONG));
+
+        execute("select routine_name, routine_body, data_type, routine_definition" +
+            " from information_schema.routines " +
+            " where routine_type = 'FUNCTION'");
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.rows()[0][0], is("substract_test"));
+        assertThat(response.rows()[0][1], is("javascript"));
+        assertThat(response.rows()[0][2], is("long"));
+        assertThat(response.rows()[0][3], is("function substract_test(a, b, c) { return a - b - c; }"));
+        execute("drop function substract_test(long, long, long)");
```

Comment:
drop function should probably be in a finally block or `@After` directive to make sure it's also executed if some assertion fails to prevent subsequent tests from failing